### PR TITLE
f-content-cards@2.2.0-beta.4 - voucher card flash

### PR DESCRIPTION
**Pull request by matthewhardern**
Adds change to provide voucher flashing transition, Have added tests and colour changes to existing branch to finish off the work for a ongoing ticket.

- [x] Unit tests have been [created|updated]

### CODE REVIEW

**Comment by ashleynolan** [comment]
Upgrade note – I've had to clean up some of our git history. If you could rebase master onto your branch and push back up to this PR, that'd be appreciated 🙌

File: packages/f-content-cards/src/components/cardTemplates/VoucherCard.vue [hunk 1]
```diff
@@ -97,4 +169,29 @@ export default {
...
+    .c-contentCard-voucher-copy-cooldownTick-leave-to {
+        opacity: 0;
+    }
+
+    .c-contentCard-voucher-copy-cooldownTick-leave-active {
+        transition: opacity 1s;
+    }
+
+    .c-contentCard-voucher-code-cooldown-tick {
+        // background-color: $color-text--success;
```
Comment (ross-gratton) [inline packages/f-content-cards/src/components/cardTemplates/VoucherCard.vue]: Should this be removed?

**Review by ross-gratton — COMMENTED** [review]
Should all the f-footer components be in this PR, is that because of Ash's rebase?

File: packages/f-content-cards/src/components/cardTemplates/VoucherCard.vue (hunk 1 above)
Comment (matthewhardern) [reply packages/f-content-cards/src/components/cardTemplates/VoucherCard.vue]: yes thanks for the spot, also sorted out the rebase mess

**Comment by matthewhardern** [comment]
> Should all the f-footer components be in this PR, is that because of Ash's rebase?

Have sorted the rebase issues, my bad

**Review by ross-gratton — DISMISSED** [review]

File: packages/f-content-cards/src/components/cardTemplates/_tests/VoucherCard.test.js [hunk 2]
```diff
@@ -71,23 +72,41 @@ describe('contentCards › VoucherCard', () => {
...
             });
+            it('should display the correct label while in cool down', async () => {
+                // Act
+                wrapper.vm.copyVoucherCode();
+                await wrapper.vm.$nextTick();
+                const copiedLabelSpan = wrapper.find('[data-test-id="contentCard-copied-label"]');
+                // Assert
+                expect(copiedLabelSpan.text()).toEqual(copy.codeCopiedLabel);
+            });
+            it('should display the correct label when cool down ends after 3 seconds', async () => {
```
Comment (kevinrodrigues) [inline packages/f-content-cards/src/components/cardTemplates/_tests/VoucherCard.test.js:107]: Really minor but could you add line breaks between the Act & Assert parts please so it's easier to read.

File: packages/f-content-cards/src/components/cardTemplates/VoucherCard.vue [hunk 3]
```diff
@@ -41,13 +64,45 @@ export default {
...
+            copyState: COPY_STATE_AVAILABLE,
+            cooldownTimeout: null,
             url
         };
     },
 
     computed: {
         isAnniversaryCard () {
             return this.cardType === 'anniversaryCard1';
+        },
```
Comment (kevinrodrigues) [inline packages/f-content-cards/src/components/cardTemplates/VoucherCard.vue]: We should be adding jsdoc comments to these methods/computed props.

File: packages/f-content-cards/src/components/cardTemplates/VoucherCard.vue [hunk 4]
```diff
@@ -14,19 +14,42 @@
...
     </card-container>
 </template>
 
 <script>
 import copyToClipboard from 'copy-to-clipboard';
+import { TickIcon } from '@justeat/f-vue-icons';
 import CardContainer from './CardContainer.vue';
 import { normaliseCardType } from './utils';
 
+const COPY_STATE_AVAILABLE = 'available';
```
Comment (kevinrodrigues) [inline packages/f-content-cards/src/components/cardTemplates/VoucherCard.vue]: Can these be pulled from a constants file?

File: packages/f-content-cards/src/components/cardTemplates/VoucherCard.vue [hunk 5]
```diff
@@ -41,13 +64,45 @@ export default {
...
+                this.$style['c-contentCard-voucher-copy'],
+                ...(this.copyState === COPY_STATE_COOLDOWN ? [
+                    this.$style['c-contentCard-voucher-copy-copied']
+                ] : []),
+                ...(this.copyState === COPY_STATE_TRANSITIONOUT ? [
+                    this.$style['c-contentCard-voucher-copy-transition-out']
+                ] : [])
+            ];
+        },
+        inCooldown () {
```
Comment (kevinrodrigues) [inline packages/f-content-cards/src/components/cardTemplates/VoucherCard.vue]: Minor, line break before this method.

File: packages/f-content-cards/src/components/cardTemplates/VoucherCard.vue (hunk 4 above)
Comment (matthewhardern) [reply packages/f-content-cards/src/components/cardTemplates/VoucherCard.vue]: have moved into seperate file

**Review by kevinrodrigues — APPROVED** [review]

**Review by ross-gratton — APPROVED** [review]

**Pull request merged**